engine/src/deploy_macosx.cpp

```diff
@@ -1491,23 +1490,24 @@ static bool MCDeployToMacOSXReadHeader(bool p_big_endian, MCDeployFileRef p_engi
             swap_load_command_hdr(p_big_endian, t_command);
 
 		// Now allocate memory for the full command record
-		r_commands[i] = (load_command *)malloc(t_command . cmdsize);
-		if (r_commands[i] == NULL)
+		if (!MCMemoryAllocate(t_command . cmdsize, t_commands[i]))
```

Comment:
I think there's a potential memory leak here.  Using the `MCAutoArray<load_command*>` will prevent the outer array from being leaked.

However, `MCAutoArray` doesn't delete each of its contained items in its destructor, so if this function returns early due to an error each of buffers allocated by this line will be leaked.

I don't know how best to solve this problem.